public/app/core/components/Select/MetricSelect.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import _ from 'lodash';
+
+import Select from './Select';
+import { Variable } from 'app/types/templates';
+
+export interface Props {
+  onChange: (value: string) => void;
+  options: any[];
+  isSearchable: boolean;
+  value: string;
+  placeholder?: string;
+  className?: string;
+  groupName?: string;
+  variables?: Variable[];
+}
+
+interface State {
+  options: any[];
+}
+
+export class MetricSelect extends React.Component<Props, State> {
+  static defaultProps = {
+    variables: [],
+    options: [],
+    groupName: 'Options',
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = { options: [] };
+  }
+
+  componentDidMount() {
+    this.setState({ options: this.buildOptions(this.props) });
+  }
+
+  componentWillReceiveProps(nextProps: Props) {
+    if (nextProps.options.length > 0 || nextProps.variables.length) {
+      this.setState({ options: this.buildOptions(nextProps) });
+    }
+  }
+
+  shouldComponentUpdate(nextProps: Props) {
+    const nextOptions = this.buildOptions(nextProps);
+    return nextProps.value !== this.props.value || !_.isEqual(nextOptions, this.state.options);
+  }
+
+  buildOptions({ variables = [], groupName = '', options }) {
+    return variables.length > 0
+      ? [
+          this.getVariablesGroup(),
```

Comment:
I not sure why group name is something that is handled inside this component. Can't the group be created here: https://github.com/grafana/grafana/blob/05d826bdcb685c78bce1e37588b76c33d2c7cd95/public/app/plugins/datasource/stackdriver/query_aggregation_ctrl.ts#L59 

so that aggOptions contains the group & options. That way this MetricSelect can support multiple groups of metrics. The current way looks like it only supports two hard coded groups (Variables, and what ever you set group name to).

Reply:
I agree! Fixed now